base-server/index.js

```diff
@@ -327,14 +333,24 @@ class BaseServer {
   }
 
   type (name, callbacks) {
-    if (typeof name === 'function') name = name.toString()
-    if (this.types[name]) {
-      throw new Error(`Action type ${name} was already defined`)
-    }
-    if (!callbacks || !callbacks.access) {
-      throw new Error(`Action type ${name} must have access callback`)
+    if (name instanceof RegExp) {
+      if (this.regexTypes[name]) {
+        throw new Error(`Action type ${name} was already defined`)
```

Comment:
It is OK to avoid this test for RegExp, since it is an advanced API (You may find that it is hard to find a regexp duplicate when you will move to `this.regexTypes: Map<RegExp, callbacks>`).